### ISSUE TITLE
[Feature] Expand Hexdump Logging Capabilities

### ISSUE THREAD

**Issue by ZackeryRSmith** [KeithBrown39423/Hexdump#20]
# [Feature]

## Feature Suggestion

Please consider adding the following things to the logging system

**Basic Execution Logs**
* command arguments passed
* file being read:
  * size in bytes
  * file basename
  * file path (absolute)

**Error Logs**
* where in the file the error occurred
* the error type (which should show if the issue is due to the file or the program itself)
* error message (should give a small description, and possibly a potential fix?)

**Info Logs**
* milestones in the file (halfway, every n bytes)
    i. helps keep track of where hexdump is looking and how close it is to finishing
* system resource usage (every so often)
    
**Warning Logs**
* (if support for large files gets added): File size (may take long to finish if it's over a certain size)
* if system resources are getting bogged down past a certain point mention that

**After Execution Logs**
* Total time took
* Peak of ram usage
* Number of warnings (if applicable)

## Use Case
When bugs occur, or working to optimize Hexdump logs can be used to help find where things are slow.

## TLDR;
Added more logging ability to Hexdump for debugging errors, and to understand what Hexdump is doing better. I only listed a few things to add, but many more should come as Hexdump expands.

**Comment by KeithBrown39423** [comment]
Here's a checklist so that you can see which things have been added so far:

Note: Anything that is struck out won't be added

- [x] Command Arguments Passed
- [x] File Name
- [x] File Size
- [x] File Path (Absolute)
- [X] Error Type
- [x] Error Message
- [x] Milestones
- [x] Warnings for Large Files
- [x] Total Execution Time
~~Error Location~~
~~System Resource Usage (at Certain Intervals)~~
~~Heavy System Resource Usage~~
~~Peak Ram Usage~~
~~Number of Warnings~~

**Comment by KeithBrown39423** [comment]
Added in 18238d1
